### DISCUSSION TITLE
[DISCUSS] Change the NettyHashTimerTimeoutManager to throw PlcTimeoutExceptions?

### DISCUSSION THREAD

**Discussion by chrisdutz** [apache/plc4x#1042] [General]
Hi all,

so I just noticed, that the NettyHashTimerTimeoutManager throws "TimeoutExceptions" if an operation takes too long.
Unfortunately this is outside the "PlcException" or "PlcRuntimeException" hierarchy and therefore falls through many PlcExcpeiton catch blocks ... I'd like to change this to throw exceptions that are children of PlcRuntimeException.

What do you think?

Chris

**Comment by sruehl** [comment]
yes, please

**Reply by sruehl** [reply]
and while at it we should search (regex) if there are any non *Plc*Exceptions are thrown

**Reply by chrisdutz** [reply]
I guess most plc connection cache issues can be tracked back to that general problem ...

**Comment by splatch** [comment]
The `TimeoutException` is a common java type for reporting timeouts, I am not sure if `CompletableFuture` have any special handling for it, but other places can actually rely on it. Personally I don't mind changing it to throw `PlcTimeoutException`, as long as it is consistent and does not confuse users I'm all for it.
